RACK-Ontology/ontology/SYSTEM.sadl

```diff
@@ -10,67 +10,68 @@
  * of the Defense Advanced Research Projects Agency (DARPA).
  */
 
+/************** edit history *****************
+  *
+  * dwa - 11 Nov 2020 - removed incorrect usages of relationships
+  * dwa - 11 Nov 2020 - added that developedBy is a type of wasAssociatedWith
+  *
+  *********************************************/
+
 uri "http://arcos.rack/SYSTEM" alias sys.
 import "http://arcos.rack/PROV-S".
 import "http://arcos.rack/DOCUMENT".
 
-// Do we want to add MTBF, type (software, hybrid etc.,) to SYSTEM? They would be needed in an assurance case.
-
 SYSTEM
 	(note "SYSTEMs classify logical parts of the whole system being developed")
 	is a type of ENTITY.
 
-	partOf (note "Used to define a relationship to a higher level system") describes SYSTEM with values of type ENTITY.
-	partOf is a type of wasDerivedFrom.
+	partOf (note "Used to define a relationship to a higher level system") describes SYSTEM with values of type SYSTEM.
 
-	producedBy (note "ACTIVITY(s) participating in developing this SYSTEM") describes SYSTEM with values of type ACTIVITY.
+	producedBy (note "ACTIVITY(s) participating in developing this SYSTEM") describes SYSTEM with values of type SYSTEM_DEVELOPMENT.
+	producedBy describes SYSTEM with at most 1 value.
 	producedBy is a type of wasGeneratedBy.
 
-	provides (note "List of entities that this SYSTEM provides to others") describes SYSTEM with values of type ENTITY.
-	provides is a type of wasDerivedFrom.
+	provides (note "List of entities that this SYSTEM provides to others") describes SYSTEM with values of type FUNCTION.
 
-	requires (note "List of entities that this SYSTEM requires from others") describes SYSTEM with values of type ENTITY.
-	requires is a type of wasDerivedFrom.
+	requires (note "List of entities that this SYSTEM requires from others") describes SYSTEM with values of type FUNCTION.
 
-	function (note "functionality / capability of a SYSTEM") describes SYSTEM with values of type FUNCTION. 
-	function is a type of wasDerivedFrom. 
+	functionalityOverview (note "functionality / capability of a SYSTEM") describes SYSTEM with values of type string. 
+	functionalityOverview describes SYSTEM with at most 1 value.
 	
 INTERFACE
-	(note "INTERFACEs describe directed connections between an ENTITY (e.g. SYSTEM) providing what another ENTITY (e.g. SYSTEM) requires")
+	(note "INTERFACEs describe directed connections between a SYSTEM providing what another SYSTEM requires")
 	is a type of ENTITY.
 
-	source (note "ENTITY (e.g. SYSTEM) that provides what the INTERFACE offers") describes INTERFACE with values of type ENTITY.
-	source is the inverse of provides.
-	source is a type of wasDerivedFrom.
+	commodity (note "the thing conveyed by the interface") describes INTERFACE with values of type string.
+	commodity describes INTERFACE with at most 1 value.
```

Comment:
I believe this is another case that we worked out with TA1 partners. It may be that eventually what goes here is a reference to an entity class (a foreign key, if you will), but so far there's not enough information to enforce that idea, and certainly not enough definition of which classes of entities might be referred to. So for now, I'd rather leave this as string, and change it only if our data model decision team decides to do that.